app/uk/gov/hmrc/agentregistrationfrontend/connectors/EmailVerificationConnector.scala

```
@@ -0,0 +1,73 @@
+import uk.gov.hmrc.agentregistrationfrontend.model.VerificationStatusResponse
+import uk.gov.hmrc.agentregistrationfrontend.model.VerifyEmailRequest
+import uk.gov.hmrc.agentregistrationfrontend.model.VerifyEmailResponse
+import uk.gov.hmrc.agentregistrationfrontend.util.RequestAwareLogging
+import uk.gov.hmrc.agentregistrationfrontend.util.RequestSupport.given
+import uk.gov.hmrc.http.HttpReads.Implicits.*
+import uk.gov.hmrc.http.HttpResponse
+import uk.gov.hmrc.http.StringContextOps
+import uk.gov.hmrc.http.client.HttpClientV2
+
+import javax.inject.Inject
+import javax.inject.Singleton
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+@Singleton
+class EmailVerificationConnector @Inject() (
+  http: HttpClientV2,
+  appConfig: AppConfig
+)(implicit val ec: ExecutionContext)
+extends RequestAwareLogging:
+
+  def verifyEmail(request: VerifyEmailRequest)(using
+    rh: RequestHeader
+  ): Future[Option[VerifyEmailResponse]] = http
+    .post(url"${appConfig.emailVerificationBaseUrl}/email-verification/verify-email")
+    .withBody(Json.toJson(request))
+    .execute[HttpResponse]
+    .map { response =>
+      response.status match {
+        case 201 => Some(response.json.as[VerifyEmailResponse])
+        case status =>
+          logger.error(s"verifyEmail error; HTTP status: $status, message: $response")
+          None
+      }
+    }
+
+  def checkEmail(credId: String)(using
+    rh: RequestHeader
+  ): Future[Option[VerificationStatusResponse]] = http
+    .get(url"${appConfig.emailVerificationBaseUrl}/email-verification/verification-status/$credId")
+    .execute[HttpResponse]
+    .map { response =>
+      response.status match {
+        case 200 => Some(response.json.as[VerificationStatusResponse])
+        case 404 => Some(VerificationStatusResponse(List.empty))
+        case status =>
+          logger.error(s"email verification status error for $credId; HTTP status: $status, message: $response")
```

Same as above, it's easier to let it fail here then propagating failure in None up to the place where we have to construct response to the browser:

Please change method signature to return
```
Future[VerificationStatusResponse]
```
and throw exception in case of unexpected response from called microservice.